sktime/transformations/series/difference.py

```diff
@@ -0,0 +1,317 @@
+
+    Difference transformations are applied at the specified lags in the order
+    provided. For example, using lags=[1, 12] corresponds to applying a
+    standard first difference, then differencing the first-differenced series
+    at lag 12 (in the event the input data has a monthly periodicity, this
+    would equate to a first difference followed by a seasonal difference).
+
+    The transformation works for univariate and multivariate timeseries. However,
+    the multivariate case applies the same differencing to every series.
+
+    Parameters
+    ----------
+    lags : int or array-like, default = 1
+        The lags used to difference the data.
+        If a single `int` value is
+
+    remove_missing : bool, default = True
+        Whether the differencer should remove the initial observations that
+        contain missing values as a result of the differencing operation(s).
+
+    Attributes
+    ----------
+    lags : np.ndarray
+        Lags used to perform the differencing of the input series.
+
+    remove_missing : bool
+        Stores whether the Differencer removes initial observations that contain
+        missing values as a result of the differencing operation(s).
+
+    Example
+    -------
+    >>> from sktime.transformations.series.difference import Differencer
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> transformer = Differencer(lags=[1, 12])
+    >>> y_hat = transformer.fit_transform(y)
+    """
+
+    _tags = {
+        "fit-in-transform": False,
+        "transform-returns-same-time-index": False,
+        "univariate-only": False,
+    }
+
+    def __init__(self, lags=1, remove_missing=True):
+        self.lags = lags
+        self.remove_missing = remove_missing
+        self._Z = None
+        self._cumulative_lags = None
+        self._prior_cum_lags = None
+        self._prior_lags = None
+        super(Differencer, self).__init__()
+
+    def _check_inverse_transform_index(self, Z):
+        """Check fitted series contains indices needed in inverse_transform."""
+        first_idx = Z.index.min()
+        orig_first_idx, orig_last_idx = self._Z.index.min(), self._Z.index.max()
+
+        is_contained_by_fitted_z = False
+        is_future = False
+
+        if first_idx < orig_first_idx:
+            msg = [
+                "Some indices of `Z` are prior to timeseries used in `fit`.",
+                "Reconstruction via `inverse_transform` is not possible.",
+            ]
+            raise ValueError(" ".join(msg))
+
+        elif Z.index.difference(self._Z.index).shape[0] == 0:
+            is_contained_by_fitted_z = True
+
+        elif first_idx > orig_last_idx:
+            is_future = True
+
+        pad_z_inv = self.remove_missing or is_future
+
+        cutoff = Z.index[0] if pad_z_inv else Z.index[self._cumulative_lags[-1]]
+        fh = ForecastingHorizon(
+            np.array([*range(-1, -(self._cumulative_lags[-1] + 1), -1)])
```

Comment:
Would that work?
```suggestion
            np.arange(-1, -(self._cumulative_lags[-1] + 1), -1)
```

Reply:
I belive it does and I've switched the code to that.